openapi/openapi_v2_backend_configuration_test.go

```diff
@@ -124,6 +124,302 @@ func TestGetHost(t *testing.T) {
+			})
+		})
+	})
+}
+
+func TestValidateRegion(t *testing.T) {
+	Convey("Given a region and a list of allowed regions", t, func() {
+		region := "allowed"
+		allowedRegions := []string{region}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(&spec.Swagger{SwaggerProps: spec.SwaggerProps{Swagger: "2.0"}}, openAPIDocumentURL)
+		Convey("When validateRegion method is called with a region that is allowed", func() {
+			err := specV2BackendConfiguration.validateRegion(region, allowedRegions)
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+		})
+		Convey("When validateRegion method is called with a non allowed region", func() {
+			err := specV2BackendConfiguration.validateRegion("nonAllowed", allowedRegions)
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "region nonAllowed not matching allowed ones [allowed]")
+			})
+		})
+	})
+}
+
+func TestGetDefaultRegion(t *testing.T) {
+	Convey("Given a specV2BackendConfiguration that is multi-region", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.${region}.some-backend.com",
+					extTfProviderRegions:         "rst1, dub1",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getDefaultRegion() method is called", func() {
+			region, err := specV2BackendConfiguration.getDefaultRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("Then region should match the default one, which is the first one found in the 'x-terraform-provider-regions' extension value", func() {
+				So(region, ShouldEqual, "rst1")
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is NOT multi-region", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getDefaultRegion() method is called", func() {
+			_, err := specV2BackendConfiguration.getDefaultRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "failed to get default region value: service is not multi-region")
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is multi-region", t, func() {
```

Comment:
mention that it doesn't have any regions as part of the description

Reply:
done